packages/@aws-cdk/cdk/lib/core/tokens.ts

```diff
@@ -29,6 +31,43 @@ export class Token {
 
         return value;
     }
+
+    /**
+     * Return a string representation of this token
+     *
+     * This string representation can be embedded into strings and
+     * will be turned into the actual representation of the token
+     * when resolve() is called on the string.
+     */
+    public toString(): string {
+        if (this.stringRepr === undefined) {
```

Comment:
If `typeof(valueOfFunction)` is a primitive (`string`, `number`, `boolean`), then just return it or it's `.toString()` representation (dah!). Basically the only situation where you need to tokenize is when `valueOrFunction` is a `function` or an `object`.